File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.py
class Solution:
    def inorder(self, root, ans):
        if root is None:
            return
        self.inorder(root.left, ans)
        ans.append(root.val)
        self.inorder(root.right, ans)

    def getMinimumDifference(self, root):
        ans = []
        self.inorder(root, ans)
        diff = sys.maxsize
        for i in range(len(ans) - 1):
            if ans[i + 1] - ans[i] < diff:
                diff = ans[i + 1] - ans[i]
        return diff

File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.cpp
class Solution {
public:
    void inorder(TreeNode*root,vector<int>&ans){
        if(root==nullptr)return;
        inorder(root->left,ans);
        ans.push_back(root->val);
        inorder(root->right,ans);
    }
    int getMinimumDifference(TreeNode* root) {
        vector<int>ans;
        inorder(root,ans);
        int diff=INT_MAX;
        for(int i=0;i<ans.size()-1;i++){
            if(ans[i+1]-ans[i]<diff)diff=ans[i+1]-ans[i];
        }
        return diff;
    }
};
